app/presenters/recent_items.rb

```diff
@@ -9,22 +9,39 @@ class RecentItems
   @@last_refresh = nil
   @@bag = nil
 
-  def initialize
-    @how_many_works_to_show = 5
-    @how_often_to_change = 60 * 10 # ten minutes in seconds.
-    @how_many_works_in_bag = 50
+  def initialize(
+    how_many_works_to_show: 5,
+    how_often_to_change:    60 * 10, # ten minutes in seconds
+    how_many_works_in_bag:  50 )
+
+    @how_many_works_to_show = how_many_works_to_show
+    @how_often_to_change = how_often_to_change 
+    @how_many_works_in_bag = how_many_works_in_bag
   end
 
   def recent_items
     @@bag = nil if time_for_a_new_bag?
     # The randomization function is seeded from @@last_refresh,
     # which is a class instance variable.
     # Hence, the shuffled selection is the same for all users at a given moment.
-    fetch_bag.shuffle(random: Random.new(@@last_refresh))[0... @how_many_works_to_show]
+    maybe_fetch_bag.shuffle(random: Random.new(@@last_refresh))[0... @how_many_works_to_show]
+  end
+
+  def fetch_bag
+    Work.where('published = true').
```

Comment:
Would be better to write this as `(published: true)`. (I know it wasn't before either).

Reply:
Fixed/